assets/js/googlesitekit/notifications/components/Notification/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ViewedStateObserver from './ViewedStateObserver';
+import { useHasBeenViewed } from '../../hooks/useHasBeenViewed';
+import useNotificationEvents from '../../hooks/useNotificationEvents';
+import { Grid, Row } from '../../../../material-components';
+
+export default function Notification( { id, children } ) {
+	const ref = useRef();
+	const viewed = useHasBeenViewed( id );
+	const trackEvents = useNotificationEvents( id );
+
+	// Track view once.
+	useEffect( () => {
+		if ( viewed ) {
+			trackEvents.view();
+		}
+	}, [ viewed, trackEvents ] );
+
+	return (
+		<section id={ id } className="googlesitekit-publisher-win" ref={ ref }>
+			<Grid>
+				<Row>{ children }</Row>
+			</Grid>
```

Comment:
Let's delegate this to the layout components. As a general rule, we should avoid splitting grid components between different components. This also gives the most flexibility to the notification and avoids the need to later add props for passing to the grid at this level.